Add DC_Multilingual support

koertho opened the pull request:
Dieser PR soll support für DC_Multilingual bringen.

Die QueryBuilder-Mechanik ist leider nicht ideal. Ich musste ein paar Dinge inkl. Interface anpassen, um überhaupt die Logik umzusetzen. 

Wir müssen uns da generell nochmal unterhalten.

koertho committed:
first working draft

koertho committed:
cleanup list item provider, use correct event dispatcher class

koertho committed:
make all correctly working, fix method name

koertho committed:
work on detail page url

koertho committed:
make reader mc compatible

koertho committed:
add listener for changelanguage lang switcher event

koertho committed:
fix copy paste error

ericges committed:
introduce ReaderRequestAttribute and RequestManager to centralize request attribute handling and enhance reader context management

ericges committed:
refactor ListView logic: replace MultilingualListViewResolver with ListViewDetailsPageUrlGeneratedEvent and associated listeners; introduce service integration for multilingual and comments bundles; enhance ListViewManager with event dispatching capability

ericges committed:
refactor MultilingualListener: rename to Terminal42ChangelanguageListener and update references accordingly

ericges committed:
update Terminal42Changelanguage integration: replace MultilingualListener with Terminal42ChangelanguageListener and adjust service definition

ericges commented on src/ListItemProvider/ListItemProviderInterface.php:
```diff
@@ -19,6 +19,7 @@ interface ListItemProviderInterface
     public function fetchCount(
         ListQueryBuilder        $listQueryBuilder,
         FilterContextCollection $filters,
+        ContentContext         $contentContext,
```
Das muss nochmal überprüft werden.

ericges committed:
refactor DcMultilingualListItemProvider: adjust method signature formatting for improved readability

ericges reviewed (changes requested):
ContentContext in Signatur prüfen. Womöglich sollte es dafür insgesamt Config-Objekte geben statt der elendig langen Signaturen.

ericges committed:
optimize Terminal42ChangelanguageListener: add pageFinder caching for improved performance and refine language-specific page resolution logic

ericges committed:
refactor Terminal42ChangelanguageListener: enhance pageFinder caching logic for language-specific page resolution

ericges committed:
refactor Terminal42ChangelanguageListener: introduce getLanguage method to centralize language retrieval logic and improve code readability

ericges committed:
refactor Terminal42ChangelanguageListener and DcMultilingualHelper: centralize language retrieval in DcMultilingualHelper and improve query handling in Terminal42ChangelanguageListener

ericges committed:
Fix AbstractPriorityServiceDescriptorRegistry: update has method signature to allow nullable key parameter

ericges committed:
update composer.json: restrict symfony dependencies to ^5.4 || ^6.0 for compatibility consistency

ericges committed:
Refactor ListItemProvider and event structure:
- Remove `DcMultilingualListItemProvider` and related dependencies.
- Introduce generic fetch events (`FetchIdsEvent`, `FetchCountEvent`, `FetchEntriesEvent`) and `FetchSubject` enum.
- Update managers (`ListViewManager`, `ReaderManager`) to leverage new event-based fetching logic.
- Simplify method signatures and improve dependency handling across affected classes.

ericges committed:
Add multilingual display field to tl_content and update ContentContainer constants

ericges committed:
Refactor ListItemProvider: streamline entry retrieval by returning combined entries directly

ericges committed:
Refactor Fetch Events and Managers:
- Replace `FetchSubject` enum with a streamlined event structure.
- Consolidate `FetchEntriesEvent` into `FetchListEntriesEvent` with enhanced configuration (`FetchSingleEntryConfig`).
- Rename and restructure `FetchIdsEvent` to `FetchAutoItemEvent`.
- Simplify filter context handling and enhance dependency organization.

ericges committed:
Add multilingual and meta configuration fields across language files and update ContentContainer constants for enhanced list and filter settings

ericges committed:
update dc_multilingual display settings and override

ericges committed:
Enhance ListViewManager: implement entry caching and refactor form action retrieval

ericges committed:
Fix README.dc_multilingual.md: correct typo in SQL update statement for dcMultilingual_display

ericges committed:
Enhance ListItemProvider and EventsListItemProvider: add detailed PHPDoc annotations for improved code documentation and clarity

ericges committed:
Enhance ListView: add detailed PHPDoc annotations for improved code documentation and clarity

ericges committed:
Refactor event structure and ListView logic:

- Replace `ListViewBuiltEvent` with `ListViewRenderEvent`.
- Replace `ReaderBuiltEvent` with `ReaderRenderEvent`.
- Introduce `AbstractFlareEvent` as a base class for events.
- Add `getEventName()` method to centralize event name management.
- Dispatch events with `getEventName()` instead of static strings.
- Refactor event-driven data handling in ListView and Reader controllers.
- Introduce `ListViewBuildEvent` and restructure `ListViewBuilder` with event dispatching.
- Adjust `PaletteEvent` to include dynamic event name resolution.
- General cleanup and improvement of dependency handling.

ericges committed:
Refactor event classes to extend AbstractFlareEvent and streamline event dispatching

ericges committed:
Document FLARE events in README.md: add details for event dispatching, including event names, classes, and descriptions.

ericges committed:
Update README.md: clarify FLARE event namespace with a note

ericges reviewed (approved):
Let's go!
